packages/compose/src/hooks/use-merge-refs/index.js

```diff
@@ -3,8 +3,12 @@
  */
 import { useRef, useCallback, useLayoutEffect } from '@wordpress/element';
 
-/** @typedef {import('@wordpress/element').RefObject} RefObject */
-/** @typedef {import('@wordpress/element').RefCallback} RefCallback */
+/* eslint-disable jsdoc/valid-types */
+/**
+ * @template T
+ * @typedef {T extends import('react').Ref<infer R> ? R : never} TypeFromRef
```

Comment:
Do we really need `never` here?

Reply:
What would you suggest in place of `never`, if the template type doesn't extend `Ref`? `never` is essentially the error case for this utility type.

Reply:
I understand, then `never` makes sense 👍